### ISSUE TITLE
`types.Row.FieldNames()` not in order

### ISSUE THREAD

**Issue by paulheg** [apache/spark-connect-go#86]
Hi, I just noticed that `FieldNames()` iterates over a `map` and therefore does not return the columns in order every time.
This creates a mismatch between `Values()` and `FieldNames()`, where they appear in different orders.
There is currently no way in getting the columns in the order they were specified in the query.

This could be fixed by sorting the map entries in the `FieldNames()` call, or trough storing the column names additionally in an array.

At least there should be a disclaimer to this!

Best regards
Paul

**Comment by grundprinzip** [comment]
Correct, I ran into this in a different context. This should be fixed properly.
